hpy/devel/include/hpy/inline_helpers.h

```diff
@@ -395,4 +395,67 @@ HPyBool_FromLong(HPyContext *ctx, long v)
     return HPyBool_FromBool(ctx, (v ? true : false));
 }
 
+/**
+ * Adjust start/end slice indices assuming a sequence of the specified length.
+ *
+ * Out of bounds indices are clipped in a manner consistent with the handling of
+ * normal slices. This function cannot fail and does not call interpreter
+ * routines.
+ *
+ * :param ctx:
+ *     The execution context.
+ * :param length:
+ *     The length of the sequence that should be assumed for adjusting the
+ *     indices.
+ * :param start:
+ *     Pointer to the start value (must not be ``NULL``).
+ * :param stop:
+ *     Pointer to the stop value (must not be ``NULL``).
+ * :param step:
+ *     The step value of the slice (must not be ``0``)
+ *
+ * :return:
+ *     Return the length of the slice. Always successful. Doesn’t call Python code.
+ */
+HPyAPI_INLINE_HELPER HPy_ssize_t
+HPySlice_AdjustIndices(HPyContext *ctx, HPy_ssize_t length, HPy_ssize_t *start, HPy_ssize_t *stop, HPy_ssize_t step)
```

Comment:
Do we have an UNUSED macro to mark `ctx`?

Reply:
we only have
```
#define _HPy_UNUSED __attribute__((unused))
```
in `hpy.h`. I'll introduce one. Since all tests are green, let me do the fixup in #415 (since that needs an update anyway).